### DISCUSSION TITLE
Postgres `session_replication_role` results in migration failure

### DISCUSSION THREAD

**Discussion by ddluc** [mikro-orm/mikro-orm#5746] [Q&A]
### Describe the bug

I'm running into an issue where I cannot run migrations on my staging or production environments due to security concerns regarding the `session_replication_role` in Postgres using Digital Ocean's managed database service. I figure this may be an issue on other managed database systems as well. 

Per Digital Ocean: 

> I understand you needed replication permission for able to run Mikro-ORM. Unfortunately, you cannot have replication privileges as this is a restriction for managed database clusters.

So, is there any recourse, or support for running migrations without replication privileges? I know there are some issues floating around that involve setting`disableForeignKeys` in the config: 

```
migrations: { disableForeignKeys: false }
```

However, this did not resolve the issue for me. 

If anyone has any thoughts or ideas, let me know!

### Reproduction

I cannot clearly reproduce the error, since it only occurs on a hosted environment with restricted privileges

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

5.8.6

### Node.js version

v20.15.0

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
> 5.8.6

Please upgrade, you might be facing something that has already been fixed, this is really old. Even if not, you won't get any fixes to old versions.

> I cannot clearly reproduce the error, since it only occurs on a hosted environment with restricted privileges

This is irrelevant, the problem is about the query you can see locally too, it's just that in your production environment it cannot be executed.

> migrations: { disableForeignKeys: false }

And this is the correct solution, without you providing any reproduction I have no idea why it does not work for you. Note that there is also `schemaGenerator: { disableForeignKeys: false }`, but if your problem is running migrations, using the option under `migrations` should be the one to use.

**Comment by suiyun39** [comment]
In the latest version of MikroORM, this issue still exists.

When `schemaGenerator.disableForeignKeys` is set to `true`, the SQL output of `mikro-orm schema:create` includes the statement:

```sql
set session_replication_role = 'replica';
```

However, not all environments (especially cloud databases or managed PostgreSQL instances) grant permission to execute this command.

**Reply by B4nan** [reply]
Note that the option is no longer set to true by default in the latest version. I am not getting this query for `schema:create` unless I explicitly enable it. Maybe you set it to `true` instead of `false`?

https://github.com/mikro-orm/mikro-orm/blob/746f43fd7ac513962c23930c9fe3ad31148a2af2/packages/core/src/utils/Configuration.ts#L126

**Reply by suiyun39** [reply]
My mistake, the actual value should be true, I mean when the foreign key is disabled, a command requiring high permissions will be generated
